src/main/scala/scalaz/nio/channels/AsynchronousByteChannel.scala

```diff
@@ -72,6 +76,61 @@ class AsynchronousByteChannel(private val channel: JAsynchronousByteChannel) {
 
 }
 
+class AsynchronousServerSocketChannel(private val channel: JAsynchronousServerSocketChannel) {
+
+  // TODO enforce the order of operations? e.g. bind andThen accept(s)?
+
+  /**
+   * Binds the channel's socket to a local address and configures the socket to listen for connections.
+   */
+  def bind(address: InetSocketAddress): IO[Exception, Unit] =
+    IO.syncException(channel.bind(address)) *> IO.unit
```

Comment:
Maybe just `IO.syncException(...).void` here?

Reply:
Yep, that’d be nicer